plugins/org.eclipse.fordiac.ide.library/src/org/eclipse/fordiac/ide/library/LibraryManager.java

```diff
@@ -673,30 +673,41 @@ public boolean belongsTo(final Object family) {
 	 * @param autoImport   if library should be automatically imported into project
 	 * @param resolve      define if dependencies should get resolved on import
 	 *                     (irrelevant if {@code autoImport} is false)
-	 * @return {@link java.net.URI} of the extracted library folder
+	 * @return {@link java.net.URI} of the extracted library folder encapsulated in
+	 *         a {@link DownloadResult}
 	 */
-	private java.net.URI libraryDownload(final String symbolicName, final VersionRange versionRange,
+	private DownloadResult<java.net.URI> libraryDownload(final String symbolicName, final VersionRange versionRange,
 			final Version preferred, final IProject project, final boolean autoImport, final boolean resolve) {
 		FordiacLogHelper.logInfo("Attempting to download library " + symbolicName + " with version " + versionRange //$NON-NLS-1$ //$NON-NLS-2$
 				+ " preferring " + preferred); //$NON-NLS-1$
 		final List<IArchiveDownloader> downloaders = TypeLibraryManager.listExtensions(DOWNLOADER_EXTENSION,
 				IArchiveDownloader.class);
-		Path archivePath;
+		DownloadResult<Path> dlResult;
+		final StringBuilder errors = new StringBuilder();
 		final VersionRange range = (versionRange == null || versionRange.isEmpty()) ? ALL_RANGE : versionRange;
 		for (final var downloader : downloaders) {
 			if (!downloader.isActive()) {
 				continue;
 			}
 			try {
-				archivePath = downloader.downloadLibrary(symbolicName, range, preferred);
-				if (archivePath != null) {
-					return extractLibrary(archivePath, project, autoImport, resolve);
+				dlResult = downloader.downloadLibrary(symbolicName, range, preferred);
+				if (dlResult.status() == DownloadResult.Status.OK) {
+					return new DownloadResult<>(extractLibrary(dlResult.result(), project, autoImport, resolve));
+				}
+				// ignore NO_CONFIG
+				if (dlResult.status() == DownloadResult.Status.NOT_FOUND
+						|| dlResult.status() == DownloadResult.Status.CONFIG_ERROR
+						|| dlResult.status() == DownloadResult.Status.ERROR) {
+					errors.append(" | "); //$NON-NLS-1$
+					errors.append(downloader.getName());
+					errors.append(": "); //$NON-NLS-1$
+					errors.append(dlResult.message());
```

Comment:
Here I am unsure if this should be done using MessageFormat

Reply:
`StringBuilder` is better in my opinion as there could be multiple downloaders involved